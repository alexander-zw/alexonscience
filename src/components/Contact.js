/**
 * This component contains my contact information.
 */
import React from 'react';
import '../styles/index.css'

const Contact = () => {
    return (
       <div className="text-div bottom-div">
            <p>Feel free to reach out to me!</p>
            <p>Email: alexwu68 [at] berkeley [dot] edu</p>
            <p>
                <a href="https://www.youtube.com/channel/UCaV0jdBmPzgBk6AYweICoMA">Youtube</a>
                &nbsp;(tinyurl.com/alexonscience)
            </p>
            <p><a href="https://github.com/alexander-zw">Github</a></p>
            <p><a href="https://www.linkedin.com/in/alexander-wu-a0145a173/">Linkedin</a></p>
       </div>
    );
}

export default Contact;
