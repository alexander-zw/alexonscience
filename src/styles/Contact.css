.social-container {
    background: #eee;
    padding: 25px 50px;
}

a.social {
    margin: 0 1rem;
    transition: transform 250ms;
    display: inline-block;
}

a.social:hover {
    transform: translateY(-2px);
}

a.youtube {
    color: #eb3223;
}

a.github {
    color: #24292e;
}

a.linkedin {
    color: #2867b2;
}
