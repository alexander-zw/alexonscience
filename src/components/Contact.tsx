import "../styles/index.css";
import "../styles/Contact.css";

import React from "react";

import { faGithub, faLinkedin, faYoutube } from "@fortawesome/free-brands-svg-icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

import MetaTags from "./subcomponents/MetaTags";

/**
 * This component contains my contact information, including some fancy social
 * media icons.
 */
function Contact() {
    const accounts = [
        {
            title: "YouTube",
            class: "youtube",
            icon: faYoutube,
            link: "https://www.youtube.com/channel/UCaV0jdBmPzgBk6AYweICoMA",
        },
        {
            title: "GitHub",
            class: "github",
            icon: faGithub,
            link: "https://github.com/alexander-zw",
        },
        {
            title: "LinkedIn",
            class: "linkedin",
            icon: faLinkedin,
            link: "https://www.linkedin.com/in/azwu",
        },
    ];

    const socialComponent = accounts.map((account, index) => (
        <a
            href={account.link}
            title={account.title}
            className={`${account.class} social`}
            key={index}
        >
            <FontAwesomeIcon icon={account.icon} size="2x" />
        </a>
    ));

    return (
        <div className="text-div bottom-margin">
            <MetaTags path="/contact" />

            <p>Feel free to reach out to me!</p>
            <p>Email: alexwu68 [at] berkeley [dot] edu</p>

            <p>Social media:</p>
            <div className="social-container">{socialComponent}</div>
        </div>
    );
}

export default Contact;
