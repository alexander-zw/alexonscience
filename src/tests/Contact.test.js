/**
 * Test for the Contact.js component.
 */
import React from "react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import Contact from "../components/Contact";

it("renders correctly", () => {
    const tree = renderer.create(<Contact />).toJSON();
    expect(tree).toMatchSnapshot();
});
