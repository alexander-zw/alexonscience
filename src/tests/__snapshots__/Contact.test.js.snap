// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="text-div bottom-div"
>
  <p>
    Feel free to reach out to me!
  </p>
  <p>
    Email: alexwu68 [at] berkeley [dot] edu
  </p>
  <p>
    <a
      href="https://www.youtube.com/channel/UCaV0jdBmPzgBk6AYweICoMA"
    >
      Youtube
    </a>
    Â (tinyurl.com/alexonscience)
  </p>
  <p>
    <a
      href="https://github.com/alexander-zw"
    >
      Github
    </a>
  </p>
  <p>
    <a
      href="https://www.linkedin.com/in/alexander-wu-a0145a173/"
    >
      Linkedin
    </a>
  </p>
</div>
`;
